TestTranspiler is flaky.

See builds https://circleci.com/gh/influxdata/platform/270 and https://circleci.com/gh/influxdata/platform/272 where nothing changed but the test failed differently. And then in https://circleci.com/gh/influxdata/platform/273 the tests passed.


```
--- FAIL: TestTranspiler (0.01s)
    --- FAIL: TestTranspiler/SELECT_mean(value),_max(value)_FROM_db0..cpu (0.00s)
    	transpiler_test.go:757: unexpected spec:root["edges"][8]["parent"]:
    			-: "mean0"
    			+: "max0"
    		root["edges"][9]["parent"]:
    			-: "max0"
    			+: "mean0"
    --- FAIL: TestTranspiler/SELECT_a_+_b_FROM_db0..cpu (0.00s)
    	transpiler_test.go:757: unexpected spec:root["edges"][4]["parent"]:
    			-: "filter0"
    			+: "filter1"
    		root["edges"][5]["parent"]:
    			-: "filter1"
    			+: "filter0"
FAIL
FAIL	github.com/influxdata/platform/query/influxql	0.011s
```

Also on my local system a `go test -count 100 ./query/influxql/` is able to reproduce the failure on master.

--- a/query/influxql/join.go
+++ b/query/influxql/join.go
@@ -58,10 +58,10 @@ func Join(t *transpilerState, cursors []cursor, on, except []string) cursor {
 		}
 	}
 
-	// Retrieve the parent tables from the tables map.
+	// Retrieve the parent ids from the cursors.
 	parents := make([]query.OperationID, 0, len(tables))
-	for id := range tables {
-		parents = append(parents, id)
+	for _, cur := range cursors {
+		parents = append(parents, cur.ID())
 	}
 	id := t.op("join", &functions.JoinOpSpec{
 		TableNames: tables,

